arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -54,30 +50,21 @@ class PerformValveIsolationTraceViewController: UIViewController {
     var selectedCategory: AGSUtilityCategory?
     var startingLocationElement: AGSUtilityElement!
     var utilityNetwork: AGSUtilityNetwork!
-    
-    /// The gas line layer ./3 and gas device layer ./0 are created from the feature service URL.
-    lazy var layers: [AGSFeatureLayer] = {
-        let urls = [
-            featureServiceURL.appendingPathComponent("3"),
-            featureServiceURL.appendingPathComponent("0")
-        ]
-        return urls.map { (url) -> AGSFeatureLayer in
-            let featureTable = AGSServiceFeatureTable(url: url)
-            return AGSFeatureLayer(featureTable: featureTable)
-        }
-    }()
+    var serviceGeodatabase: AGSServiceGeodatabase!
+    var layers = [AGSFeatureLayer]()
     
     // MARK: Initialize map and utility network
     
     /// Create a map.
     ///
     /// - Parameter layers: The feature layers for the utility network.
-    /// - Returns: An `AGSMap` object.
-    func makeMap(with layers: [AGSFeatureLayer]) -> AGSMap {
+    func setMap(with layers: [AGSFeatureLayer]) {
         let map = AGSMap(basemapStyle: .arcGISStreetsNight)
         // Add the utility network feature layers to the map for display.
         map.operationalLayers.addObjects(from: layers)
-        return map
+        // Create the utility network, referencing the map.
+        utilityNetwork = AGSUtilityNetwork(url: featureServiceURL, map: map)
```

Comment:
Hmm it seems to be an odd place to initialize the `utilityNetwork` property... Leave it to Phil's judgement.

Reply:
I forwarded this question to Jennifer N. from the utility networks team:

> Yup that is correct, using UN constructor with map parameter will add this UN to the map's UtilityNetworks[]collection. If you use UN constructor with just URL, before loading it, you can add to map.UtilityNetworks property.. but once loaded, you cannot add to a map anymore. This map and UN relationship allows the tables on map to be used by UN - sharing the same instance.

Reply:
Thanks!